Sources/SimilaritySearchKit/Core/Embeddings/Models/NativeContextualEmbeddings.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  NativeContextualEmbeddings.swift
+//  
+//
+//  Created by Zach Nagengast on 10/11/23.
+//
+
+import Foundation
+import NaturalLanguage
+import CoreML
+
+#if canImport(NaturalLanguage.NLContextualEmbedding)
+@available(macOS 14.0, iOS 17.0, *)
+public class NativeContextualEmbeddings: EmbeddingsProtocol {
+    public let model: ModelActor
+    public let tokenizer: any TokenizerProtocol
+
+    // Initialize with a language
+    public init(language: NLLanguage = .english) {
+        self.tokenizer = NativeTokenizer()
+        guard let nativeModel = NLContextualEmbedding(language: language) else {
+            fatalError("Failed to load the Core ML model.")
+        }
+        Self.initializeModel(nativeModel)
+        self.model = ModelActor(model: nativeModel)
+    }
+
+    // Initialize with a script
```

Comment:
btw tangentially do you know whether the different scripts eg cjk can be be compared across each other somehow, or are they totally incompatible? in terms of finding similarity across languages that span these diff sets. thanks

Reply:
I'm not sure about scripts but give it a try and report back! Apple tried to emphasize that this is "multilingual" so should be pretty close.